src/test/java/org/openrewrite/java/testing/junit6/MinimumJreConditionsTest.java

```diff
@@ -0,0 +1,1072 @@
+                  import org.junit.jupiter.api.condition.JRE;
+
+                  import static org.junit.jupiter.api.condition.JRE.*;
+
+                  class MyTest {
+                      @Test
+                      @DisabledOnJre(%s)
+                      void testNotOnJava8() {
+                          System.out.println("Not Java 8");
+                      }
+                  }
+                  """.formatted(jre),
+                """
+                  import org.junit.jupiter.api.Test;
+
+                  class MyTest {
+                      @Test
+                      void testNotOnJava8() {
+                          System.out.println("Not Java 8");
+                      }
+                  }
+                  """
+              )
+            );
+        }
+
+        @Test
+        void keepAnnotationValuesForCurrentOrNewerJREs() {
+            rewriteRun(
+              java(
+                """
+                  import org.junit.jupiter.api.Test;
+                  import org.junit.jupiter.api.condition.DisabledOnJre;
+
+                  class MyTest {
+                      @Test
+                      @DisabledOnJre(versions = { 8, 17, 21 })
+                      void testNotOnJava17() {
+                          System.out.println("Not Java 17 or 21");
+                      }
+                  }
+                  """,
+                """
+                  import org.junit.jupiter.api.Test;
+                  import org.junit.jupiter.api.condition.DisabledOnJre;
+
+                  class MyTest {
+                      @Test
+                      @DisabledOnJre(versions = {17, 21})
+                      void testNotOnJava17() {
+                          System.out.println("Not Java 17 or 21");
+                      }
+                  }
+                  """
+              )
+            );
+        }
+
+        @ParameterizedTest
+        @ValueSource(strings = {
+          "JRE.JAVA_17",
+          "value = JRE.JAVA_17",
+          "versions = 17",
+          "JAVA_17",
+          "JRE.JAVA_21",
+          "value = JRE.JAVA_21",
+          "versions = 21",
+          "JAVA_21"
+        })
+        void keepDisabledOnCurrentOrNewerJRE(String jre) {
+            rewriteRun(
+              java(
+                """
+                  import org.junit.jupiter.api.Test;
+                  import org.junit.jupiter.api.condition.DisabledOnJre;
+                  import org.junit.jupiter.api.condition.JRE;
+
+                  import static org.junit.jupiter.api.condition.JRE.*;
+
+                  class MyTest {
+                      @Test
+                      @DisabledOnJre(%s)
+                      void testNotOnJava17() {
+                          System.out.println("Not Java 17");
+                      }
+                  }
+                  """.formatted(jre)
+              )
+            );
+        }
+
+        @CsvSource({
```

Comment:
I'm somewhat surprised at seeing changes even when we're not adding/removing values. Would it make sense to delegate to that single element annotation visitor only from visiting a changed compilation unit as well?